crates/core_arch/src/wasm/memory.rs

```diff
@@ -1,13 +1,22 @@
 #[cfg(test)]
 use stdarch_test::assert_instr;
 
+#[cfg(target_arch = "wasm32")]
 extern "C" {
     #[link_name = "llvm.wasm.memory.grow.i32"]
     fn llvm_memory_grow(mem: i32, pages: i32) -> i32;
     #[link_name = "llvm.wasm.memory.size.i32"]
     fn llvm_memory_size(mem: i32) -> i32;
 }
 
+#[cfg(target_arch = "wasm64")]
+extern "C" {
+    #[link_name = "llvm.wasm.memory.grow.i64"]
+    fn llvm_memory_grow(mem: i32, pages: i64) -> i64;
+    #[link_name = "llvm.wasm.memory.size.i64"]
+    fn llvm_memory_size(mem: i32) -> i64;
+}
```

Comment:
Would it make sense to change these from `i32`/`i64` to using `isize` instead? That way we wouldn't need to have separate versions of these intrinsics, and separate versions of the code which uses these intrinsics.

Reply:
i assumed the explicit use of `i32` was for a reason (I couldn't figure out why though). if that isn't the case I can definitely change it.

Reply:
Stdarch matches the C versions, even when it doesn't make much sense. For example on x86 i32 is often used even if the argument obviously must not be negative. Or simd masks are i32 when they could fit in u8.

Reply:
In C, these intrinsics are [defined](https://github.com/llvm/llvm-project/blob/master/clang/include/clang/Basic/BuiltinsWebAssembly.def#L25) with [size_t](https://github.com/llvm/llvm-project/blob/master/clang/include/clang/Basic/Builtins.def#L30). This would correspond to `usize` in Rust.

`memory.grow` can fail and return `-1`, but with the C intrinsics, this is interpreted as `SIZE_MAX`, which corresponds to `usize::MAX` in Rust.